### ISSUE TITLE
Remove unused encoders

### ISSUE THREAD

**Issue by scottpurdy** [numenta/nupic-legacy#1479]
Remove the following encoders:
- [x] ArithmeticEncoder - I wrote this but it isn't clear that we want to encourage adaptive encoders
- [ ] AdaptiveScalarEncoder - RDSE is better
- [x] SDRRandomEncoder - don't understand utility of this
- [x] NonUniformScalarEncoder - again, not sure we want to encourage adaptive encoders
- [x] ScalarSpaceEncoder
- [x] VectorEncoder - just wraps multiple encoders
- [x]  UtilityEncoder - functionality is outside the scope of encoders

**Comment by rhyolight** [comment]
@scottpurdy ping

**Comment by scottpurdy** [comment]
@rhyolight What is the action item for me?

**Comment by rhyolight** [comment]
@scottpurdy Remove the AdaptiveScalarEncoder? I figured you weren't done, since you assigned to yourself and did all but one task.

**Comment by scottpurdy** [comment]
@rhyolight Right, yes that is still remaining. I'll want to check with people to make sure the alternatives are sufficient. Not a priority right now.
